presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DesugaringOptimizer.java

```diff
@@ -113,6 +149,44 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<Void> context)
                     originalConstraint);
         }
 
+        @Override
+        public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
+        {
+            PlanNode left = context.rewrite(node.getLeft());
+            PlanNode right = context.rewrite(node.getRight());
+            return new JoinNode(
+                    node.getId(),
+                    node.getType(),
+                    left,
+                    right,
+                    node.getCriteria(),
+                    node.getOutputSymbols(),
+                    node.getFilter().map(this::desugar),
```

Comment:
can you extract `filter` expression as variable? That way it will be easier to notice that `filter` is rewritten.